R/DataModel.R

```diff
@@ -310,73 +322,100 @@ naToZero <- function(x) {
 #' @param cdmSourceFile  File contained within zip that references databaseId field (used for purging data)
 #'
 #' @export
-uploadResults <- function(connectionDetails = NULL,
+uploadResults <- function(connection = NULL,
+                          connectionDetails = NULL,
                           schema,
-                          zipFileName,
+                          resultsFolder,
                           tablePrefix = "",
                           forceOverWriteOfSpecifications = FALSE,
                           purgeSiteDataBeforeUploading = TRUE,
                           cdmSourceFile = "cdm_source_info.csv",
+                          runCheckAndFixCommands = FALSE,
                           tempFolder = tempdir(),
                           specifications) {
-  start <- Sys.time()
-  connection <- DatabaseConnector::connect(connectionDetails)
-  on.exit(DatabaseConnector::disconnect(connection), add = TRUE)
-
-  unzipFolder <- tempfile("unzipTempFolder", tmpdir = tempFolder)
-  dir.create(path = unzipFolder, recursive = TRUE)
-  on.exit(unlink(unzipFolder, recursive = TRUE), add = TRUE)
-
-  ParallelLogger::logInfo("Unzipping ", zipFileName)
-  zip::unzip(zipFileName, exdir = unzipFolder)
-
-  if (purgeSiteDataBeforeUploading) {
-    database <-
-      readr::read_csv(file = file.path(unzipFolder, cdmSourceFile),
-                      col_types = readr::cols())
-    colnames(database) <-
-      SqlRender::snakeCaseToCamelCase(colnames(database))
-    databaseId <- database$databaseId
+  if (is.null(connection)) {
+    if (!is.null(connectionDetails)) {
+      connection <- DatabaseConnector::connect(connectionDetails)
+      on.exit(DatabaseConnector::disconnect(connection))
+    } else {
+      stop("No connection or connectionDetails provided.")
+    }
   }
 
-  uploadTable <- function(tableName) {
-    ParallelLogger::logInfo("Uploading table ", tableName)
+  if (connection@dbms == "sqlite" & schema != "main") {
+    stop("Invalid schema for sqlite, use schema = 'main'")
+  }
 
-    primaryKey <- specifications %>%
-      dplyr::filter(tableName == !!tableName &
-                      tolower(primaryKey) == "yes") %>%
-      dplyr::select("fieldName") %>%
-      dplyr::pull()
+  start <- Sys.time()
 
-    if (purgeSiteDataBeforeUploading &&
-      "database_id" %in% primaryKey) {
+  # TODO: Seems specific to CD? What if the cdmSourceFile does not exist?
```

Comment:
The `cdmSourceFile` will not exist for results created by Strategus and this seemed potentially some functionality specific to CohortDiagnostics. Perhaps the databaseId is something that could be passed to this function?

Reply:
The db ID isn't necessarily known by the user at this point. It's also incredibly convenient to be able to trivially remove data and re-insert without having to jump through hoops.

For example, if you run a study with a parameter that is wrong on 15 databases, you change it and then re-run having to manually do this delete becomes a real pain. If there isn't a file which contains the database_ids that you're removing then I think this could be skipped with 

`if (purgeSiteDataBeforeUploading && file.exists(cdmSourceFile))
....
`

Reply:
I'm thinking that in the case of Strategus, the caller could set the `cdmSourceFile` to the path to the DatabaseMetaData folder which contains the `database_meta_data.csv` which should satisfy our needs here. I'll just need to add a check such that if the cdmSourceFile is set to a relative path, it is assumed to be in the resultsFolder otherwise it will use the path passed by the user.

Reply:
Also happy for the "database_meta_data.csv" file to be the default filename.